File: src/main/scala/blog/typed/scaladsl/Worker.scala
package blog.typed.scaladsl

import akka.actor.typed.Behavior
import akka.actor.typed.scaladsl.Actor

object Worker {
  sealed trait Command
  final case class Job(payload: String) extends Command

  val workerBehavior: Behavior[Command] =
    Actor.immutable[Command] { (ctx, msg) =>
      msg match {
        case Job(payload) =>
          println(s"Worker ${ctx.self} got job $payload")
          Actor.same
      }
    }
}

File: src/main/java/blog/typed/javadsl/Worker.java
package blog.typed.javadsl;

import akka.actor.typed.Behavior;
import akka.actor.typed.javadsl.Actor;

public abstract class Worker {
  private Worker() {
  }

  interface Command {
  }

  public static class Job implements Command {
    public final String payload;

    public Job(String payload) {
      this.payload = payload;
    }
  }

  public static Behavior<Command> behavior() {
    return Actor.immutable(Command.class)
      .onMessage(Job.class, (ctx, msg) -> {
        System.out.println("Worker " + ctx.getSelf() + " got job " + msg.payload);
        return Actor.same();
      })
      .build();
  }

}